pkg/registry/generic/etcd/etcd.go

```diff
@@ -245,13 +248,22 @@ func (e *Etcd) Update(ctx api.Context, obj runtime.Object) (runtime.Object, bool
 			}
 			return obj, nil
 		}
+
 		creating = false
+		newVersion, err := e.Helper.ResourceVersioner.ResourceVersion(obj)
```

Comment:
Why is this Update function using the AtomicUpdate pattern? Shouldn't it just use CompareAndSwap?